scripts/release/build/image_build_process.py

```diff
@@ -9,10 +9,64 @@
 from python_on_whales.exceptions import DockerException
 
 from lib.base_logger import logger
+from scripts.release.branch_detection import get_cache_scope, get_current_branch
 
 DEFAULT_BUILDER_NAME = "multiarch"  # Default buildx builder name
 
 
+def ensure_ecr_cache_repository(repository_name: str, region: str = "us-east-1"):
```

Comment:
[non blocking]

I would like some comments to state a few things that might seem obvious, but that I understood fully because I had to implement some caching in a pipeline recently:

- We create a (or multiple) repo to store our image caches, and for each build docker will cache from/to there.
- They are cached in OCI format
- Why we need to do per branch caching